diagram_analysis/diagram_generator.py

```
@@ -451,32 +478,403 @@ def generate_analysis(self) -> Path:
+        root_analysis: AnalysisInsights,
+        sub_analyses: dict[str, AnalysisInsights],
+        trace_result: TraceResult,
+        agent_llm: BaseChatModel,
+        parsing_llm: BaseChatModel,
+    ) -> None:
+        """Patch impacted sub-analyses using EASE-encoded JSON patches."""
+        from diagram_analysis.analysis_patcher import merge_patched_sub_analyses, patch_sub_analysis
+
+        patched: dict[str, AnalysisInsights] = {}
+        seen_parents: set[str] = set()
+        patch_tasks: list[tuple[str, AnalysisInsights, ImpactedComponent]] = []
```

Ruff (0.15.7)

[error] Undefined name `ImpactedComponent`

(F821)